src/otx/algorithms/common/adapters/torch/dataloaders/samplers/otx_sampler.py

```diff
@@ -0,0 +1,98 @@
+"""OTX sampler."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import math
+
+from torch.utils.data import Dataset
+from torch.utils.data.sampler import Sampler
+
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.common.utils.task_adapt import unwrap_dataset
+
+logger = get_logger()
+
+
+class OTXSampler(Sampler):  # pylint: disable=too-many-instance-attributes
```

Comment:
Could you add unit tests for this class?

Reply:
done @https://github.com/openvinotoolkit/training_extensions/pull/2428/commits/a496a82861ea094fca937a3c06caff9aac85f3fd